xx.md

```diff
@@ -0,0 +1,153 @@
+
+Public:
+```js
+{
+  "kind": 30050,
+  "pubkey": "<user-pubkey>",
+  "tags": [
+	["d", "<pubkey-of-friend>"],
+	["p", "<pubkey-of-friend>"],
+	["p", "<optional-predecessor-pubkey-of-friend>"],
+	["attestation", JSONStringify({
+	  "pubkey": "<pubkey-of-friend>"
+	  "tags": [<attested-to-tags>],
+	  "content": {
+		"<attested-key>": "<attested-value>",
+		"<attested-key>": "<attested-value>"
+	  }
+	})]
+  ],
+  "content": ""
+}
+```
+
+Private:
+```js
+  "kind": 30050,
+  "pubkey": "<user-pubkey>",
+  "tags": [
+	["d", "<encrypted-and-hashed-pubkey"]
+  ],
+  "content": Nip44Encrypt(JSONStringify([
+	["p", "<pubkey-of-friend>"],
+	["attestation", JSONStringify({
+	  "pubkey": "<pubkey-of-friend>"
+	  "tags": [<attested-to-tags>],
+	  "content": {
+		"<attested-key>": "<attested-value>",
+		"<attested-key>": "<attested-value>"
+	  }
+	})]
+  ]))
+```
+
+* For a _public_ attestation:
+  * The `d` tag and a `p` tag MUST include a public key for the attested to metadata.
+  * Another `p` tag SHOULD be included if there was a predecessor public key. This helps to inform other users of a link between the predecessor public and a successor public key.
+  * The `attestations` tag MUST include JSON stringified copy of the attested to event pubkey, tags and content of the `kind 0` event. It is a partial copy of the `kind 0` event.
+* For a _private_ attestation:
+  * The `d` tag MUST be an encrypted and hashed version of the public key (hex encoding of a sha256 hash of an encrypted, with NIP-44, of the public key).
+  * The `p`, `metadata` and `attestations` tags, as the same as the public attestation, MUST be JSON stringified and NIP-44 encrypted in the content field.
+
+## Migration Keys
+
+This is a new field on a users metadata (`kind 0`) event with a key of `migration_keys`. Its purpose is to define a set of migration keys that can be used to help migrate to a successor key in the future. The event can assign anywhere from `1` to `n` migration keys. A threshold number of keys (`m` of `n`) can be assigned to verify this event.
+
+The value should be as follows:
+
+```js
+[<threshold>, <migration-pubkey-1>, <migration-pubkey-2>]
+```
+
+* The array SHOULD be all strings, including the threshold.
+* Clients MAY present a user interface to make an attestation, if this field is available on the metadata.
+* Clients MAY use hardware devices and NIP-06 seed phrases to store the migration keys.
+
+### Revocation Event Signing
+
+A _Key Revocation Event_ MAY be signed with `m` of `n` of these migration keys. This can help assist the migration process for those that have previously attested to the `migration_keys`.
+
+The following `migration-sigs` tag MAY be included:
+```js
+{
+	"kind": 50,
+	"tags": [
+		// other tags
```

Comment:
```suggestion
		// other tags...
```

Reply:
I'll look at updating some of the formatting in another update. Some of the examples could also be expanded and improved. :)